core/app/models/spree/promotion_code/batch_builder.rb

```diff
@@ -39,13 +39,15 @@ def generate_random_codes
       new_codes = Array.new(max_codes_to_generate) { generate_random_code }.uniq
       codes_for_current_batch = get_unique_codes(new_codes)
 
-      codes_for_current_batch.each do |value|
+      codes_for_current_batch = codes_for_current_batch.map do |value|
```

Comment:
collecting results of large code counts can exceed available memory pretty quickly.